training/models/training_action.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# © 2015 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+
+from openerp import _, api, fields, models
+from .common import M
+from .. import exceptions as ex
+
+
+class ActionType(models.Model):
+    """Types of training actions.
+
+    Depending on the training action type, a training action may expect some
+    type of hours. For example:
+
+    - If a training action's type is "on-site", it expects on-site hours.
+    - If its type is "online", it expects online hours.
+    - If its type is "mixed", it expects both on-site and online hours.
+
+    You can configure it as you wish.
+    """
+    _name = M % "action_type"
+    _sql_constraints = [("unique_name",
+                         "UNIQUE(name)",
+                         "Name must be unique.")]
+
+    name = fields.Char(required=True, index=True, translate=True)
+    action_ids = fields.One2many(
+        M % "action",
+        "type_id",
+        "Training actions",
+        help="Training actions of this type.")
+    expected_duration_type_ids = fields.Many2many(
+        M % "duration_type",
+        string="Expected hour types",
+        help="These types of hours are expected in this type of training "
+             "action. For example, a training of type 'mixed' may expect "
+             "hours of types 'on-site' and 'online'.")
+
+
+class Action(models.Model):
+    """Define training actions.
```

Comment:
Yes, I put it like this to distinguish between what defines a course (currently training action) and what uses a course (currently training group), but being training groups events after all, your suggestion seems better. I'll change it.